Sources/MeiliSearch/Model/SearchResult.swift

```diff
@@ -16,10 +17,71 @@ public struct SearchResult<T>: Codable, Equatable where T: Codable, T: Equatable
     /// Number of documents taken.
     public let limit: Int
 
+    /// Total number of matches,
+    public let nbHits: Int
+
+    /// Whether `nbHits` is exhaustive.
+    public let exhaustiveNbHits: Bool?
+
+    /// Distribution of the given facets.
+    public let facetsDistribution: [String: [String: Int]]?
```

Comment:
Can't you use a custom type `Facet` that is also Codable to avoid the Behemot? 🙂

```swift
public let facetsDistribution: [String: [Facet]]?
```

Reply:
I don't think it's possible to map this, especially knowing that the structure is like this:

```swift
"genres": [
    "Classic Regency nove": 1,
    "High fantasy‎": 1,
    "Fantasy": 2,
    "Novel": 2,
    "Bildungsroman": 1
]
```

By creating the `Facet` type, we need to know beforehand the keys used in this JSON tree, since the JSON is dynamic, this is not possible.